### PR TITLE
Teach the renderer to keep thread alive if engine requests it

### PR DESCRIPTION
**Pull request by miniksa**
Teaches renderer base to keep thread alive if engine requests it.
`DxEngine` now requests it if shaders are on.

- The render engine interface now has a true/false to return whether the
  specific renderer wants another frame to immediately follow up. The
  renderer base will ask for this information as it ends the paint on
  any particular engine (which is the time where invalid regions are
  typically cleaned up) and just poke the render thread the same as if
  an invalidation request came in from outside of render-land. That will
  trigger the render thread to just keep moving in the same way as any
  other invalidation.

## Validation Steps Performed
- [x] Actually built it
- [x] Actually try it

I promised this in #8994

### CODE REVIEW

File: src/renderer/base/renderer.cpp [hunk 1]
```diff
@@ -133,6 +133,14 @@ try
 
     auto endPaint = wil::scope_exit([&]() {
         LOG_IF_FAILED(pEngine->EndPaint());
+
+        // If the engine tells us it really wants to redraw immediately,
+        // tell the thread so it doesn't go to sleep and ticks again 
+        // at the next opportunity.
+        if (pEngine->RequiresContinuousRedraw())
```
Comment (zadjii-msft) [inline src/renderer/base/renderer.cpp:140]: oh my gosh this is so clever, I love it

**Review by zadjii-msft — APPROVED** [review]
This is GREAT. I don't think there's a way to be sure if a shader is actually using the `Time` variable or not - we'd somehow need to inspect the compiled shader to see if the variable was ever referenced? There's no way that's possible. 

Lets :shipit:

**Comment by miniksa** [comment]
### TODO
- [x] See if we can turn this off if a shader doesn't use time variable or at least for the in-built retro one?
   - The answer to this one is that we can turn it off for the in-built one, but I cannot see a way to detect if the shader is actually using the time parameter. Perhaps in the future if this becomes an extension point instead of just a set-path-to-HLSL-file, we can have someone declare they need the timer tick to differentiate. Or someone will have another idea of how the HLSL can return that info in the future.

**Comment by DHowett** [comment]
nit: reword title to be imperative; pretend it completes the sentence "this commit will"

**Comment by DHowett** [comment]
Moved from commit body

- [x] `samples/PixelShaders/Animate_breathe.hlsl` breathe
- [x] `samples/PixelShaders/Animate_scan.hlsl` scanline

**Review by DHowett — APPROVED** [review]

**Comment by DHowett** [comment]
Clever as.

**Pull request merged**